### PR TITLE
Dynamic CS Form

### PR DESCRIPTION
**Pull request by kien-ngo**
Changes:
- [x] Add different forms for different use-cases
  - [x] Connect
  - [x] Engine
  - [x] Contracts
  - [x] Account
  - [x] Other 
- [x] Message sent to Unthread will contain all the metadata from the form
- [ ] Forms now allow users to upload attachments
- [x] chore: Abstract some code

Internal discussion: PROD-692

---

## PR-Codex overview
The focus of this PR is to enhance the support ticket creation process by adding different forms for specific problem areas and improving markdown content generation.

### Detailed summary
- Added support for different problem areas in support forms
- Enhanced markdown content generation for support tickets
- Created specific form components for account, engine, and other support forms

> The following files were skipped due to too many changes: `src/components/help/contact-forms/connect/AffectedAreaInput.tsx`, `src/components/help/contact-forms/connect/index.tsx`, `src/components/help/contact-forms/contracts/index.tsx`, `src/components/help/contact-support-modal.tsx`

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

**Comment by jnsdls** [comment]
#### 📝 Changed routes:


| [**/support**](https://thirdweb-lggy8ebjg.thirdweb-preview.com/support) |
|:-:|
|<a href="https://thirdweb-lggy8ebjg.thirdweb-preview.com/support">Screenshot of /support</a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/4uAWW9GGS2iUd27hupWjeZJ0GHew4yEYehwFm7keB9rLteJ_YE0gKz7Ar2Q2CeARITZ6NeSLamDyyyoq9TKbGg.png">(view full size)</a> |







Commit e820e0a1672814c1dda2bd5bee457ab86c921606 (https://thirdweb-lggy8ebjg.thirdweb-preview.com).

File: src/pages/support.tsx [hunk 1]
```diff
@@ -8,14 +8,17 @@ import {
...
 import { NextSeo } from "next-seo";
+import dynamic from "next/dynamic";
 import { PageId } from "page-id";
 import { FiArrowRight } from "react-icons/fi";
 import { Card, Heading, Link, LinkButton, Text } from "tw-components";
 import { ThirdwebNextPage } from "utils/types";
 
+const ContactSupportModal = dynamic(
+  () => import("components/help/contact-support-modal"),
+);
```
Comment (jnsdls) [inline src/pages/support.tsx:21]: what's the benefit of lazy loading this?

File: src/components/help/contact-support-modal.tsx [hunk 2]
```diff
@@ -22,26 +22,109 @@ import {
...
+    problemAreas: [],
+  },
+  {
+    label: "Others",
+    problemAreas: [],
+  },
 ];
 
-export const ContactSupportModal = () => {
+export default function ContactSupportModal() {
```
Comment (jnsdls) [inline src/components/help/contact-support-modal.tsx:95]: generally prefer named exports over default exports -> I know this is required for the dynamic import (see separate question about that)

**Review by jnsdls — COMMENTED** [review]
some questions for now :)

File: src/pages/support.tsx (hunk 1 above)
Comment (kien-ngo) [reply src/pages/support.tsx:21]: oops 💀 I meant to lazy load the other component (https://github.com/thirdweb-dev/dashboard/pull/2556/commits/c16a96a7bc8895e9457319c44e47b1da04bc11d3)

I will revert this

File: src/components/help/contact-forms/shared/SupportForm_AttachmentUploader.tsx [hunk 3]
```diff
@@ -0,0 +1,252 @@
+// import { CreateTicketInput } from "@3rdweb-sdk/react/hooks/useApi";
```
Comment (jnsdls) [inline src/components/help/contact-forms/shared/SupportForm_AttachmentUploader.tsx:1]: if its not ready just delete the file for now

**Review by jnsdls — APPROVED** [review]
overall good just remove the empty file

File: src/components/help/contact-forms/shared/SupportForm_AttachmentUploader.tsx (hunk 3 above)
Comment (kien-ngo) [reply src/components/help/contact-forms/shared/SupportForm_AttachmentUploader.tsx:1]: thanks!

**Pull request merged**